src/contracts/version/Registry.ts

```diff
@@ -67,12 +86,32 @@ export class Registry extends Contract {
   }
 
   public registerExchangeAdapter(from: Address, args: ExchangeAdapterInformation) {
-    return this.createTransaction('registerExchangeAdapter', from, [
+    const method = 'registerExchangeAdapter';
+    const methodArgs = [
       args.exchange,
       args.adapter,
       args.takesCustody,
       args.sigs.map(sig => hexToBytes(utf8ToHex(sig))),
-    ]);
+    ];
+
+    const validate = async () => {
+      let message;
+
+      const exchangeInfo = await this.getExchangeInformation(args.adapter);
+      if (exchangeInfo.exists) {
+        message = `Adapter already exists. Adapter exists: ${exchangeInfo.exists}`;
+        throw new RegisteredEntityError(exchangeInfo.exists, message);
+      }
+
+      const adapters = await this.getRegisteredExchangeAdapters();
+      const maxRegisteredEntities = await this.getMaxRegisteredEntities();
+      if (maxRegisteredEntities.isLessThanOrEqualTo(adapters.length)) {
+        message = `Exchange limit reached. Adapters: ${adapters.length}, Limit: ${maxRegisteredEntities.toNumber()}`;
```

Comment:
The message does not need to contain the parameters. You are already passing them to the error class.